src/coreclr/interop/trackerobjectmanager.cpp

```diff
@@ -265,47 +126,15 @@ namespace
     }
 }
 
-HRESULT TrackerObjectManager::OnIReferenceTrackerFound(_In_ IReferenceTracker* obj)
+HRESULT TrackerObjectManager::HasReferenceTrackerManager()
 {
-    _ASSERTE(obj != nullptr);
-    if (s_TrackerManager != nullptr)
-        return S_OK;
-
-    // Retrieve IReferenceTrackerManager
-    HRESULT hr;
-    ComHolder<IReferenceTrackerManager> trackerManager;
-    RETURN_IF_FAILED(obj->GetReferenceTrackerManager(&trackerManager));
-
-    ComHolder<IReferenceTrackerHost> hostServices;
-    RETURN_IF_FAILED(g_HostServicesInstance.QueryInterface(IID_IReferenceTrackerHost, (void**)&hostServices));
-
-    // Attempt to set the tracker instance.
-    if (InterlockedCompareExchangePointer((void**)&s_TrackerManager, trackerManager.p, nullptr) == nullptr)
-    {
-        (void)trackerManager.Detach(); // Ownership has been transferred
-        RETURN_IF_FAILED(s_TrackerManager->SetReferenceTrackerHost(hostServices));
-    }
-
-    return S_OK;
+    return (s_TrackerManager != nullptr) ? S_OK : S_FALSE;
 }
 
-HRESULT TrackerObjectManager::AfterWrapperCreated(_In_ IReferenceTracker* obj)
+HRESULT TrackerObjectManager::TryRegisterReferenceTrackerManager(_In_ IReferenceTrackerManager* manager)
```

Comment:
Should this just return `bool` so that we are not converting between HRESULT and bool? 

Dtto for `HasReferenceTrackerManager`.